pypeit/scripts/run_to_calibstep.py

```diff
@@ -0,0 +1,100 @@
+"""
+Script to run to a single calibration step for an input frame
+
+.. include common links, assuming primary doc root is up one directory
+.. include:: ../include/links.rst
+"""
+
+from pypeit.scripts import scriptbase
+
+class RunToCalibStep(scriptbase.ScriptBase):
+
+    # TODO: Combining classmethod and property works in python 3.9 and later
+    # only: https://docs.python.org/3.9/library/functions.html#classmethod
+    # Order matters.  In python 3.9, it would be:
+    #
+    # @classmethod
+    # @property
+    #
+    # Because we're not requiring python 3.9 yet, we have to leave this as a
+    # classmethod only:
+    @classmethod
+    def name(cls):
+        """
+        Return the name of the executable.
+        """
+        return 'pypeit_run_to_calibstep'
+
+    @classmethod
+    def get_parser(cls, width=None):
+        import argparse
+
+        parser = super().get_parser(description='Run PypeIt to a single calibration step for an input frame',
+                                    width=width, formatter=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument('pypeit_file', type=str,
+                            help='PypeIt reduction file (must have .pypeit extension)')
+        parser.add_argument('frame', type=str, help='Raw science frame to reduce as listed in your PypeIt file, e.g. b28.fits.gz')
```

Comment:
Would it make sense to allow the users to provide a calibration group or a science file?  I understand why you're using the science file (it defines the fitstbl row and therefore the calibration group), but can we just provide the calibration group directly?  It feels odd that we're requiring users provide a specific science frame for this script.

Related, can we also allow the user to specify the detector or mosaic?